impeller/aiks/picture.cc

```diff
@@ -3,11 +3,23 @@
 // found in the LICENSE file.
 
 #include "impeller/aiks/picture.h"
+#include <memory>
 
 #include "impeller/entity/entity.h"
+#include "impeller/renderer/render_target.h"
 
 namespace impeller {
 
-//
+std::shared_ptr<Image> Picture::RenderToImage(AiksContext& renderer) {
+  auto coverage = pass->GetElementsCoverage();
+  auto target = RenderTarget::CreateOffscreen(
+      *renderer.context_, ISize(coverage->GetRight(), coverage->GetBottom()));
+
+  if (!renderer.Render(*this, target)) {
+    return nullptr;
+  }
+
+  return std::make_shared<Image>(target.GetRenderTargetTexture());
```

Comment:
Check that the texture is not null. Add a validation log if it is and return nullptr.

Reply:
IMO, we should, by convention, not return invalid resources out of factories that return resources via a pointer. Instead, favoring returning null. This makes the caller not have to do a null and IsValid invocation pair.

Reply:
Done.